src/main/java/community/whatever/onembackendjava/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,22 @@
+package community.whatever.onembackendjava.exception;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<Map<String, String>> handleIllegalArgumentException(IllegalArgumentException ex){
+        Map<String, String> errorResponse = new HashMap<>();
+        errorResponse.put("error", "INVALID_ARGUMENT");
+        errorResponse.put("message", ex.getMessage());
+
+        return ResponseEntity.badRequest().body(errorResponse);
+    }
+
+}
```

Comment:
```suggestion
@RestControllerAdvice
public class GlobalExceptionHandler {

    @ResponseStatus(HttpStatus.BAD_REQUEST)
    @ExceptionHandler(IllegalArgumentException.class)
    public Map<String, String> handleIllegalArgumentException(IllegalArgumentException ex) {
        return Map.of(
                "error", "INVALID_ARGUMENT",
                "message", ex.getMessage()
        );
    }
}
```

`nit` map 을 그대로 반환하는 것 보다는 dto 를 반환하는 것을 추천 합니다.
- fyi. org.springframework.web.ErrorResponse

Reply:
해당 내역 map 대신 dto를 반환하도록 수정했습니다.
dto는 간결한 작성과 유지보수를 고려하여 record 클래스로 생성했습니다.